### PR TITLE
Add distributor bucket space to ideal state operation

### PR DESCRIPTION
**Pull request by toregge**
@geirst, @vekterli: please review

### CODE REVIEW

File: storage/src/vespa/storage/distributor/operations/idealstate/idealstateoperation.cpp [hunk 1]
```diff
@@ -26,6 +27,7 @@ const uint32_t IdealStateOperation::MAINTENANCE_MESSAGE_TYPES[] =
 
 IdealStateOperation::IdealStateOperation(const BucketAndNodes& bucketAndNodes)
         : _manager(NULL),
+          _bucketSpace(NULL),
```
Comment (vekterli) [inline storage/src/vespa/storage/distributor/operations/idealstate/idealstateoperation.cpp]: Prefer `nullptr`

**Review by vekterli — APPROVED** [review]
LGTM. See single comment on `nullptr` in constructor.

Later, when it's time for cleaning things up, we should probably consolidate more bucket DB functionality in the bucket space. Current design has a spread of responsibility between the distributor component and the bucket space. E.g. `_manager->getDistributorComponent().removeNodeFromDB()` versus `_bucketSpace->getBucketDatabase().remove()` (these don't do exactly the same thing, but the point remains).

**Review by vekterli — APPROVED** [review]
:+1:

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**